### PR TITLE
Fix decode nested array infinite loop and add tests

### PR DESCRIPTION
**Pull request by leoneparise**
This PR fixes the nested array decoding and adds a test case

### CODE REVIEW

File: Sources/Contentful/Decodable.swift [hunk 1]
```diff
@@ -17,11 +17,11 @@ public extension Decoder {
 
     /// The `TimeZone` the `Decoder` is using to offset dates by.
     /// Set through `ClientConfiguration`.
-    public var timeZone: TimeZone? {
+    var timeZone: TimeZone? {
```
Comment (mariobodemann) [inline Sources/Contentful/Decodable.swift]: I am not sure if changing the access control from public to internal is needed by the array nesting issue.

Do you mind helping me understand why we need this change?

File: Tests/ContentfulTests/JSONDecodingTests.swift [hunk 2]
```diff
@@ -71,6 +71,38 @@ class JSONDecodingTests: XCTestCase {
         XCTAssertEqual((wrapper.dict["array"] as! [Any])[1] as? Double, 3.3)
         XCTAssertNil(wrapper.dict["key1"])
     }
+    
+    func testHandlingNestedArrayValues() {
```
Comment (mariobodemann) [inline Tests/ContentfulTests/JSONDecodingTests.swift]: :heart_eyes:

File: Sources/Contentful/Decodable.swift (hunk 1 above)
Comment (leoneparise) [reply Sources/Contentful/Decodable.swift]: Decodable is a public interface, Swift compiler is complaining that `public` modifier is unnecessary.

**Comment by mariobodemann** [comment]
Okay, one last thing before we merge: Can we change the CHANGELOG.md and README.md to be now the version 4.2.5.

**Comment by leoneparise** [comment]
@mariobodemann Done. :)

File: CHANGELOG.md [hunk 3]
```diff
@@ -10,7 +10,7 @@ This project adheres to [Semantic Versioning](http://semver.org/) starting from
 ## Table of contents
 
 #### 4.x Releases
-- `4.2.0` Releases - [4.2.0](#420) | [4.2.1](#421) | [4.2.2](#422) | [4.2.3](#423) | [4.2.4](#424)
+- `4.2.0` Releases - [4.2.0](#420) | [4.2.1](#421) | [4.2.2](#422) | [4.2.3](#423) | [4.2.4](#424) | [4.2.5](#425)
```
Comment (mariobodemann) [inline CHANGELOG.md]: Thanks for updating those! Are the blocks down below in the readme (line 43+) updated automatically, or would we need to add them manually?

**Comment by matelo** [comment]
hello @leoneparise, is this PR waiting to be merged? If so, can you please take a look at conflict files? thanks

**Comment by matelo** [comment]
resolved in:
https://github.com/contentful/contentful.swift/pull/295

**Pull request closed without merging**